src/ProductBundle/Admin/ProductAdmin.php

```diff
@@ -84,6 +70,8 @@ public function getProductType()
      * Returns the product class from the provided request.
      *
      * @return string
+     *
+     * @deprecated since 2.2, will be removed in 3.0
```

Comment:
It is not enough with this, you need also the trigger error

Reply:
and please add a testcase which expects the deprecation

Reply:
Done